### PR TITLE
perf: Change return type of custom protocol handler from `Vec<u8>` to `Cow<'static, [u8]>`, closes #796

### LINKED ISSUE #796: Allow custom  protocol handlers to return static resources without heap allocation
**Is your feature request related to a problem? Please describe.**

The served files such as HTML/CSS/JavaScript are embedded in my application binary like below.

```rust
const SCRIPT: &[u8] = include_bytes!("bundled.js");
const LOGO: &[u8] = include_bytes!("logo.png");
```

And those resources are returned in [a custom protocol handler](https://docs.rs/wry/latest/wry/webview/struct.WebViewBuilder.html#method.with_custom_protocol) as if the response for HTTP request. Currently, custom protocol handlers return `Response<Vec<u8>>` so the body payload must be allocated on heap.

```rust
let webview = WebViewbuilder;;new(window)?
    .with_custom_protocol("MyApp".into(), move |request| {
      let path = request.uri().path();
      let body = match path {
          "/" => SCRIPT.to_vec(), // Unnecessary heap allocation
          "/logo.png" => LOGO.to_vec(), // Unnecessary heap allocation
          _ => fs::read(convert_to_filepath(path))?,
      };

      let mime = ...;

      Response::builder()
        .header(CONTENT_TYPE, mime)
        .body(body)
        .map_err(Into::into)
    });
```

However, static resources like `SCRIPT` or `LOGO` (both are a few MB) are not necessary to be allocated on heap.

In my environment, allocating a single 1~2MB file took about 3ms in release build. When the number of files grows, the overhead will increase.

**Describe the solution you'd like**

I'd like to change the return type `Response<Vec<u8>>` to `Response<Cow<'static, [u8]>>` so that static resources are not necessary to be cloned on heap memory. It can return both `&'static [u8]` and `Vec<u8>`.

- Before: `Fn(&Request<Vec<u8>>) -> Result<Response<Vec<u8>>> + 'static`
- After: `Fn(&Request<Vec<u8>>) -> Result<Response<Cow<'static, [u8]>>> + 'static`

The above example can be rewritten as follows with this change:

```rust
let webview = WebViewbuilder;;new(window)?
    .with_custom_protocol("MyApp".into(), move |request| {
      let path = request.uri().path();
      let body = match path {
          "/" => SCRIPT.into(), // Same as Cow::Borrowed(SCRIPT)
          "/logo.png" => LOGO.into(), // Same as Cow::Borrowed(LOGO)
          _ => fs::read(convert_to_filepath(path))?.into(), // Same as Cow::Owned(...)
      };

      let mime = ...;

      Response::builder()
        .header(CONTENT_TYPE, mime)
        .body(body)
        .map_err(Into::into)
    });
```

The drawback of this solution is that this breaks current users' code base. `.into()` is necessary to convert `Vec<u8>` into `Cow<'static, [u8]>`.

**Describe alternatives you've considered**

Trivial alternative would be continuing to use the current signature not to break current users' code base. And current signature is simpler.

**Would you assign yourself to implement this feature?**
- [x] Yes
- [ ] No

**Additional context**
Nothing.

### PR DESCRIPTION
**Pull request by rhysd**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [x] Yes, and the changes were approved in issue #796
- [ ] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/wry/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information

I confirmed this change using [the custom protocol example](https://github.com/tauri-apps/wry/blob/dev/examples/custom_protocol.rs) on the following platforms:

- Windows 10
- macOS 11.7
- Ubuntu 22.04

I don't have Android and iOS development environment so I could not confirm this patch on it.

### CODE REVIEW

File: examples/custom_protocol.rs [hunk 1]
```diff
@@ -27,21 +29,19 @@ fn main() -> wry::Result<()> {
...
-          "custom_protocol_page1.html"
-        } else {
-          // remove leading slash
-          &path[1..]
-        },
-      ))?)?;
+      let content = if path == "/" {
+        PAGE1_HTML.into()
+      } else {
+        // 1.. for removing leading slash
```
Comment (amrbashir) [inline examples/custom_protocol.rs]:
```suggestion
        // `1..` for removing leading slash
```

File: examples/menu.rs [hunk 2]
```diff
@@ -43,21 +45,19 @@ fn main() -> wry::Result<()> {
...
-          "custom_protocol_page1.html"
-        } else {
-          // remove leading slash
-          &path[1..]
-        },
-      ))?)?;
+      let content = if path == "/" {
+        PAGE1_HTML.into()
+      } else {
+        // 1.. for removing leading slash
```
Comment (amrbashir) [inline examples/menu.rs]:
```suggestion
        // `1..` for removing leading slash
```

**Review by amrbashir — DISMISSED** [review]
Thank you! LGTM! regardless of the few nitpicks.

**Comment by amrbashir** [comment]
for android, you can just install `rustup target add aarch64-linux-android` and then run `cargo check --target aarch64-linux-android` to make sure it compiles fine.

**Comment by rhysd** [comment]
> for android, you can just install rustup target add aarch64-linux-android and then run cargo check --target aarch64-linux-android to make sure it compiles fine.

Ah, I didn't know that checking non-host target is so easy with cargo. Thank you. I'll check the compile errors and your review points tonight.

**Comment by rhysd** [comment]
@amrbashir @wusyong I fixed Android build and resolved review comments. I apologize that I could not take time to work on this yesterday.

**Review by amrbashir — APPROVED** [review]
LGTM! Thank you.

**Pull request merged**